azure-servicebus/src/main/java/com/microsoft/azure/servicebus/management/NamespaceInfoSerializer.java

```diff
@@ -0,0 +1,104 @@
+                TRACE_LOGGER.debug("XML which failed to parse: \n %s", xml);
+            }
+        }
+
+        throw new ServiceBusException(false, "Unable to deserialize the content.");
+    }
+
+    private static NamespaceInfo parseFromEntry(Node xEntry) {
+        NamespaceInfo namespaceInfo = new NamespaceInfo();
+        NodeList nList = xEntry.getChildNodes();
+        for (int i = 0; i < nList.getLength(); i++) {
+            Node node = nList.item(i);
+            if (node.getNodeType() == Node.ELEMENT_NODE) {
+                Element element = (Element)node;
+                switch(element.getTagName())
+                {
+                    case "content":
+                        NodeList nsInfoNodes = element.getFirstChild().getChildNodes();
+                        for (int j = 0; j < nsInfoNodes.getLength(); j++)
+                        {
+                            node = nsInfoNodes.item(j);
+                            if (node.getNodeType() == Node.ELEMENT_NODE) {
+                                element = (Element) node;
+                                switch (element.getTagName())
+                                {
+                                    case "CreatedTime":
+                                        namespaceInfo.setCreatedAt(Instant.parse(element.getFirstChild().getNodeValue()));
+                                        break;
+                                    case "ModifiedTime":
+                                        namespaceInfo.setModifiedAt(Instant.parse(element.getFirstChild().getNodeValue()));
+                                        break;
+                                    case "Name":
+                                        namespaceInfo.setName(element.getFirstChild().getNodeValue());
+                                        break;
+                                    case "Alias":
+                                        namespaceInfo.setAlias(element.getFirstChild().getNodeValue());
+                                        break;
+                                    case "NamespaceType":
+                                        String namespaceType = element.getFirstChild().getNodeValue();
+                                        switch (namespaceType)
+                                        {
+                                            case "Mixed":
+                                                namespaceInfo.setNamespaceType(NamespaceType.Mixed);
+                                                break;
+                                            case "Messaging":
+                                                namespaceInfo.setNamespaceType(NamespaceType.ServiceBus);
+                                                break;
+                                            default:
+                                                namespaceInfo.setNamespaceType(NamespaceType.Unknown);
+                                                break;
```

Comment:
You can save this trouble.

Reply:
This is not a big issue. As an when a new namespaceType or a new NamespaceSKU gets released (which won't happen often), we can release a new client. But I dont think just returning string for that rare scenario is worth it.